### PR TITLE
Sales tax using oop

### PR DESCRIPTION
**Pull request by dajianosh**

### CODE REVIEW

**Review by SagarSSolanki — CHANGES_REQUESTED** [review]
Refactor code into multiple files and import them here.

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]
@dajianosh Avoid the use of global/class variables.

File: sales_tax_using_oop.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$LOAD_PATH << '.'
```
Comment (SagarSSolanki) [inline sales_tax_using_oop.rb]: There is one more way to import file without using **$LOAD_PATH << '.'** try it.

File: sales_tax_using_oop.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+require 'user_input.rb'
+require 'tax_calculations.rb'
+
+#main
+i_obj = Input.new           #object of Input Class
```
Comment (SagarSSolanki) [inline sales_tax_using_oop.rb]: object / variable name is not descriptive, do not use short form

File: sales_tax_using_oop.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+tax = Tax.new               #object of Tax Class
+details = i_obj.user_input
```
Comment (SagarSSolanki) [inline sales_tax_using_oop.rb]: **details** is also less descriptive

File: sales_tax_using_oop.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+$LOAD_PATH << '.'
+
+require 'user_input.rb'
+require 'tax_calculations.rb'
+
+#main
+i_obj = Input.new           #object of Input Class
+tax = Tax.new               #object of Tax Class
```
Comment (SagarSSolanki) [inline sales_tax_using_oop.rb]: These lines of code can be reduced!

File: tax_calculations.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+class Tax
+  def import_tax(product_name, product_price) #checking imported or not and returning import tax if applicable
```
Comment (SagarSSolanki) [inline tax_calculations.rb]: If functions are only called inside the class make them private.

File: tax_calculations.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    for item in details
+      temp = item.split
+      quantity = Float(temp[0]) 
+      product_price = Float(temp.last)
+      product_name = temp.slice(1, temp.length - 3)
+  
+      i_tax = import_tax(product_name, product_price)
+      s_tax = sales_tax(product_name, product_price)
+  
+      total_tax = total_tax + s_tax + i_tax
```
Comment (SagarSSolanki) [inline tax_calculations.rb]: use  **+=**  here

File: user_input.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class Input
+  def user_input
+    puts "Enter product details:"
+    input = Array.new
+    while(product_details = gets.chomp) != 'exit'
+      input.push(product_details) 
+    end 
+    return input
```
Comment (SagarSSolanki) [inline user_input.rb]: remove **return** keyword from here.
check how to return values in ruby functions.

**Pull request merged**